src/main/java/world/jnc/invsync/config/Config.java

```diff
@@ -139,5 +149,39 @@ static String urlEncode(String str) {
         return URLEncoder.encode(str, UTF_8);
       }
     }
+
+    @ConfigSerializable
+    @Getter
+    public static class PostgreSQL {
+      @Setting private String host = "localhost";
+      @Setting private int port = 5432;
+      @Setting private String database = "invsync";
+      @Setting private String user = "invsync";
+      @Setting private String password = "sup3rS3cur3Pa55w0rd!";
+
+      @Setting(comment = "Prefix for the plugin tables")
+      private String tablePrefix = "invsync_";
+
+      @Setting(
+        value = "postgreProperties",
+        comment =
+            "Additional properties that can be passed into the postgresql connection string. Refer to https://jdbc.postgresql.org/documentation/head/connect.html"
+      )
+      private Map<String, String> properties = new HashMap<>();
+
+      public String getUserEncoded() {
+        return urlEncode(getUser());
+      }
+
+      public String getPasswordEncoded() {
+        return urlEncode(getPassword());
+      }
+
+      @VisibleForTesting
+      @SneakyThrows(UnsupportedEncodingException.class)
+      static String urlEncode(String str) {
+        return URLEncoder.encode(str, UTF_8);
+      }
```

Comment:
Might make sense to move that to the container class.  
Not a dealbreaker if you don't.